### DISCUSSION TITLE
Advanced Infinite Scrolling Example

### DISCUSSION THREAD

**Discussion by erandow** [remix-run/remix#4180] [Q&A]
Hi everybody, 
I read the [advanced infinite scrolling](https://github.com/remix-run/remix/blob/main/examples/infinite-scrolling/app/routes/offset/advanced.tsx) example and I understand that the method that was implemented is excellent in front-end but not efficient in the backend (API). 
The approach is that you will request to get the data for a **limited** number from your backend(API) from a **starting** point, for example, in the first query you will get the data from index 0 to index 200 (start: 0, limit: 200). Then after scrolling down and for the second request, we will get the data from index 80 to 280 (start: 80, limit: 200). 

This scenario shows that the data we get from the query is redundant and will pressure the server(API) side (especially when the server is slow).
what is your idea? Am I wrong?

**Comment by kiliman** [comment]
That example always replaces the data that is rendered. So as you scroll it needs to refetch data that is at the ends of the virtual list. 

Another option is to get the initial data from `useLoaderData` and store that in local state. Then as you scroll `useFetcher` to get the next page of data and append it to local state. Perhaps drop some of the extra data if needed.

**Comment by erandow** [comment]
Thank you @kiliman, 
I think the problem is with fetching redundant data and I couldn't understand why it is labeled as an advanced example. 
As I mentioned before, the implemented example is fetching redundant data multiple times.

This is a screenshot of the network layer while scrolling the advanced example.

![Screenshot from 2022-09-18 10-17-47](https://user-images.githubusercontent.com/38919232/190887803-53d69a7a-f7ca-4710-9389-ab419846ef90.png)

As you see, we are downloading data from index 80 to 280 in the first request, we also download the data from index 160 to 360 in the second request. 
Data from index 160 to 280 are redundant.
